data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqssource/SqsSource.java

```diff
@@ -4,32 +4,103 @@
+import org.opensearch.dataprepper.plugins.aws.sqs.common.handler.SqsMessageHandler;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.metrics.SqsMetrics;
+import org.opensearch.dataprepper.plugins.aws.sqs.common.model.SqsOptions;
 import org.opensearch.dataprepper.plugins.source.sqssource.config.SqsSourceConfig;
+import org.opensearch.dataprepper.plugins.source.sqssource.handler.RawSqsMessageHandler;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 @DataPrepperPlugin(name = "sqs", pluginType = Source.class,pluginConfigurationType = SqsSourceConfig.class)
 public class SqsSource implements Source<Record<Event>> {
 
+    static final long INITIAL_DELAY = Duration.ofSeconds(20).toMillis();
+
+    static final long MAXIMUM_DELAY = Duration.ofMinutes(5).toMillis();
+
+    static final double JITTER_RATE = 0.20;
+
     private final SqsSourceConfig sqsSourceConfig;
 
+    private final AcknowledgementSetManager acknowledgementSetManager;
+
+    private final PluginMetrics pluginMetrics;
+
+    private final boolean acknowledgementsEnabled;
+
+    private final org.opensearch.dataprepper.aws.api.AwsCredentialsSupplier awsCredentialsSupplier;
+
+    private final List<ExecutorService> executorServices;
+
     @DataPrepperPluginConstructor
-    public SqsSource(final SqsSourceConfig sqsSourceConfig) {
+    public SqsSource(final PluginMetrics pluginMetrics,
+                     final SqsSourceConfig sqsSourceConfig,
+                     final AcknowledgementSetManager acknowledgementSetManager,
+                     final AwsCredentialsSupplier awsCredentialsSupplier) {
         this.sqsSourceConfig = sqsSourceConfig;
+        this.acknowledgementSetManager = acknowledgementSetManager;
+        this.acknowledgementsEnabled = sqsSourceConfig.getAcknowledgements();
+        this.pluginMetrics = pluginMetrics;
+        this.awsCredentialsSupplier = awsCredentialsSupplier;
+        this.executorServices = new ArrayList<>(sqsSourceConfig.getQueues().getUrls().size());
     }
 
     @Override
     public void start(Buffer<Record<Event>> buffer) {
         if (buffer == null) {
             throw new IllegalStateException("Buffer is null");
         }
+
+        final SqsMetrics sqsMetrics = new SqsMetrics(pluginMetrics);
+
+        SqsClient sqsClient = ClientFactory.createSqsClient(sqsSourceConfig.getAws().getAwsRegion(),
+                sqsSourceConfig.getAws().getAwsStsRoleArn(),
+                sqsSourceConfig.getAws().getAwsStsHeaderOverrides(),
+                awsCredentialsSupplier);
+
+        final Backoff backoff = Backoff.exponential(INITIAL_DELAY, MAXIMUM_DELAY).withJitter(JITTER_RATE)
+                .withMaxAttempts(Integer.MAX_VALUE);
+        SqsService sqsService = new SqsService(sqsMetrics,sqsClient,backoff);
```

Comment:
Use final keywords where ever applicable.

Reply:
added final all required places and its available for review.